lib/types/winston.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:equatable/equatable.dart';
+
+class Winston extends Equatable {
+  final BigInt _amount;
+
+  @override
+  List<Object?> get props => [_amount];
+
+  Winston(BigInt amount) : _amount = amount {
+    if (amount < BigInt.zero || !amount.isValidInt) {
```

Comment:
The getter isValidInt would return false if the value is bigger than the max safe integer value, it is not a "is integer" check.
The values of BigInt instances are already integer.

```suggestion
    if (amount < BigInt.zero) {
```